src/test/java/com/amazon/ion/impl/macro/MacroCompilerTest.kt

```diff
@@ -36,79 +36,89 @@ class MacroCompilerTest {
             signature = emptyList(),
             body = listOf(DecimalValue(emptyList(), BigDecimal("3.141592653589793")))
         ),
         "(macro cardinality_test (x?) x)" shouldCompileTo TemplateMacro(
             signature = listOf(Parameter("x", Tagged, ParameterCardinality.ZeroOrOne)),
-            body = listOf(Variable(0))
+            body = listOf(VariableRef(0))
         ),
         "(macro cardinality_test (x!) x)" shouldCompileTo TemplateMacro(
             signature = listOf(Parameter("x", Tagged, ParameterCardinality.ExactlyOne)),
-            body = listOf(Variable(0))
+            body = listOf(VariableRef(0))
         ),
         "(macro cardinality_test (x+) x)" shouldCompileTo TemplateMacro(
             signature = listOf(Parameter("x", Tagged, ParameterCardinality.OneOrMore)),
-            body = listOf(Variable(0))
+            body = listOf(VariableRef(0))
         ),
         "(macro cardinality_test (x*) x)" shouldCompileTo TemplateMacro(
             signature = listOf(Parameter("x", Tagged, ParameterCardinality.ZeroOrMore)),
-            body = listOf(Variable(0))
+            body = listOf(VariableRef(0))
         ),
         // Outer 'values' call allows multiple expressions in the body
         // The second `values` is a macro call that has a single argument: the variable `x`
         // The `literal` call causes the third (inner) `(values x)` to be an uninterpreted s-expression.
         """(macro literal_test (x) (values (values x) (literal (values x))))""" shouldCompileTo TemplateMacro(
             signature = listOf(Parameter("x", Tagged, ParameterCardinality.ExactlyOne)),
             body = listOf(
-                MacroInvocation(ByName("values"), startInclusive = 0, endInclusive = 5),
-                MacroInvocation(ByName("values"), startInclusive = 1, endInclusive = 2),
-                Variable(0),
-                SExpValue(emptyList(), startInclusive = 3, endInclusive = 5),
+                MacroInvocation(ByName("values"), selfIndex = 0, endExclusive = 6),
+                MacroInvocation(ByName("values"), selfIndex = 1, endExclusive = 3),
+                VariableRef(0),
+                SExpValue(emptyList(), selfIndex = 3, endExclusive = 6),
                 SymbolValue(emptyList(), FakeSymbolToken("values", -1)),
                 SymbolValue(emptyList(), FakeSymbolToken("x", -1)),
             ),
         ),
         "(macro each_type () (values null true 1 ${"9".repeat(50)} 1e0 1d0 2024-01-16T \"foo\" (literal bar) [] (literal ()) {} {{}} {{\"\"}} ))" shouldCompileTo TemplateMacro(
             signature = emptyList(),
             body = listOf(
-                MacroInvocation(ByName("values"), 0, 14),
+                MacroInvocation(ByName("values"), 0, 15),
                 NullValue(emptyList(), IonType.NULL),
                 BoolValue(emptyList(), true),
-                IntValue(emptyList(), 1),
+                LongIntValue(emptyList(), 1),
```

Comment:
🗺️ Another change that's not particularly noticeable in `Expression.kt`. I thought that `LongIntValue` and `BigIntValue` was a little more descriptive, and then I was able to use `IntValue` as a sealed supertype of those two.